cli/src/cli_util.rs

```diff
@@ -3101,70 +3100,79 @@ pub fn compute_commit_location(
-    let after_commits = resolve_revisions(insert_after)?;
-    let before_commits = resolve_revisions(insert_before)?;
-
-    let (new_parents, new_children) = match (destination_commits, after_commits, before_commits) {
-        (Some(destination_commits), None, None) => (destination_commits, vec![]),
-        (None, Some(after_commits), Some(before_commits)) => (after_commits, before_commits),
-        (None, Some(after_commits), None) => {
-            let new_children: Vec<_> =
-                RevsetExpression::commits(after_commits.iter().ids().cloned().collect_vec())
-                    .children()
-                    .evaluate(workspace_command.repo().as_ref())?
+    let destination_commit_ids = resolve_revisions(destination)?;
+    let after_commit_ids = resolve_revisions(insert_after)?;
+    let before_commit_ids = resolve_revisions(insert_before)?;
+
+    let (new_parent_ids, new_children_ids) =
+        match (destination_commit_ids, after_commit_ids, before_commit_ids) {
+            (Some(destination_commit_ids), None, None) => (destination_commit_ids, vec![]),
+            (None, Some(after_commit_ids), Some(before_commit_ids)) => {
+                (after_commit_ids, before_commit_ids)
+            }
+            (None, Some(after_commit_ids), None) => {
+                let new_children_ids: Vec<_> =
+                    RevsetExpression::commits(after_commit_ids.iter().cloned().collect_vec())
+                        .children()
+                        .evaluate(workspace_command.repo().as_ref())?
+                        .iter()
+                        .try_collect()?;
+
+                (after_commit_ids, new_children_ids)
+            }
+            (None, None, Some(before_commits)) => {
+                // Not using `RevsetExpression::parents` here to persist the order of parents
+                // specified in `before_commits`.
+                let new_parents = before_commits
                     .iter()
-                    .commits(workspace_command.repo().store())
-                    .try_collect()?;
-
-            (after_commits, new_children)
-        }
-        (None, None, Some(before_commits)) => {
-            // Not using `RevsetExpression::parents` here to persist the order of parents
-            // specified in `before_commits`.
-            let new_parent_ids = before_commits
-                .iter()
-                .flat_map(|commit| commit.parent_ids().iter().cloned())
-                .unique();
-            let new_parents: Vec<_> = new_parent_ids
-                .map(|commit_id| workspace_command.repo().store().get_commit(&commit_id))
-                .try_collect()?;
+                    .map(|commit_id| {
+                        workspace_command
+                            .repo()
+                            .store()
+                            .get_commit(commit_id)
+                            .map(|commit| commit.parent_ids().iter().cloned().collect_vec())
```

Comment:
nit: `.to_vec()`

but it's probably better to `.try_collect()` a list of `Commit`s first.